corgidrp/detector.py

```diff
@@ -520,7 +528,204 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+        	
+    	Returns:
+        	full_qe (np.array): ideal flat field
+        	full_residuals (np.array): flat field from uranus or neptune images
+        	percens_resel (np.array): percentage error of the flat field
+    """
+    
+    n = 420
+    full_qe=np.random.normal(1,.03,(n,n))
+    full_residuals = np.zeros((n,n))
+    rad1 = 1.26; rad4 = 1.75
+    aperture1 = CircularAperture((np.ceil(rad1), np.ceil(rad1)), r=rad1)
+    aperture4 = CircularAperture((np.ceil(rad4), np.ceil(rad4)), r=rad4)
+    mask1= aperture1.to_mask().data
+    mask4= aperture4.to_mask().data
+    rad = 50
+    for i in np.arange(len(residual_images)):
+        nx = np.arange(0,residual_images[i].shape[1])
+        ny = np.arange(0,residual_images[i].shape[0])
+        nxx,nyy = np.meshgrid(nx,ny)
+        nrr = np.sqrt((nxx-rad-5)**2 + (nyy-rad-5)**2)
+        nrr_copy = nrr.copy()
+        nrr_copy[nrr<rad] = residual_images[i][nrr<rad] 
+        nrr_copy[nrr>=rad] = None 
+        ymin = int(cent[i][0])
+        ymax = int(cent[i][1])
+        xmin = int(cent[i][2])
+        xmax = int(cent[i][3])
+
+        bool_innotzero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        bool_iniszero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]==0)
+        bool_outisnotzero = np.logical_and(nrr>=rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        
+        full_residuals[ymin:ymax,xmin:xmax][bool_innotzero] = (nrr_copy[bool_innotzero] + full_residuals[ymin:ymax,xmin:xmax][bool_innotzero])/2
+        full_residuals[ymin:ymax,xmin:xmax][bool_iniszero] = nrr_copy[bool_iniszero]
+        full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero] = full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero]
+
+    if band == 'band4':
+        full_qe_resel = ndimage.convolve(full_qe, mask4)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask4)
+        percens_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel)*100)
+    
+    elif band== 'band1':
+        full_qe_resel = ndimage.convolve(full_qe, mask1)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask1)
+        percens_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel)*100)
+    
+    full_residuals[full_residuals==0] = None
+    nx = np.arange(0,percens_resel.shape[1])
+    ny = np.arange(0,percens_resel.shape[0])
+    nxx,nyy = np.meshgrid(ny,nx)
+    nrr = np.sqrt((nxx-n/2)**2 + (nyy-n/2)**2)
+    full_residuals[nrr>165]= None
+    percens_resel[nrr>165] = None
+    full_qe[nrr>165]= None
+    full_qe=np.pad(full_qe, ((302,302),(302,302)), mode='constant',constant_values=(np.nan,))
+    full_residuals=np.pad(full_residuals, ((302,302),(302,302)), mode='constant',constant_values=(np.nan,))
+    percens_resel=np.pad(percens_resel, ((302,302),(302,302)), mode='constant',constant_values=(np.nan,))
+    
+    return (full_qe,full_residuals,percens_resel)
+    
+    
+def create_onsky_flatfield(filedir):
```

Comment:
This looks like the "step function" to make the flat field. The input of this function must be a `Dataset` object. It cannot be a file directory (Assume that the fits files have already be read in as a Dataset).